File: public/examples/integrations/toolkits/google/calendar/create_event_example_call_tool.py
from arcadepy import Arcade

client = Arcade()  # Automatically finds the `ARCADE_API_KEY` env variable

USER_ID = "{arcade_user_id}"
TOOL_NAME = "GoogleCalendar.CreateEvent"

auth_response = client.tools.authorize(
    tool_name=TOOL_NAME,
    user_id=USER_ID,
)

if auth_response.status != "completed":
    print(f"Click this link to authorize: {auth_response.url}")

# Wait for the authorization to complete
client.auth.wait_for_completion(auth_response)

tool_input = {
    "summary": "1:1 with John Doe",
    "description": "Discuss project updates and next steps",
    "start_datetime": "2024-12-31T10:00:00",
    "end_datetime": "2024-12-31T10:30:00",
    "attendee_emails": ["john.doe@example.com"],
    "calendar_id": "primary",
}

response = client.tools.execute(
    tool_name=TOOL_NAME,
    input=tool_input,
    user_id=USER_ID,
)
print(response)

File: public/examples/integrations/toolkits/outlook_calendar/create_event_example_call_tool.js
import { Arcade } from "@arcadeai/arcadejs";

const client = new Arcade(); // Automatically finds the `ARCADE_API_KEY` env variable

const USER_ID = "{arcade_user_id}";
const TOOL_NAME = "OutlookCalendar.CreateEvent";

// Start the authorization process
const authResponse = await client.tools.authorize({tool_name: TOOL_NAME});

if (authResponse.status !== "completed") {
  console.log(`Click this link to authorize: ${authResponse.url}`);
}

// Wait for the authorization to complete
await client.auth.waitForCompletion(authResponse);

const toolInput = {
  "subject": "Team Meeting",
  "body": "Discuss project updates and next steps.",
  "start_date_time": "2025-04-25T13:00:00",
  "end_date_time": "2025-04-25T14:00:00",
  "location": "Conference Room A",
  "attendee_emails": [
    "alice@example.com",
    "bob@example.com"
  ],
  "is_online_meeting": true
};

const response = await client.tools.execute({
  tool_name: TOOL_NAME,
  input: toolInput,
  user_id: USER_ID,
});

console.log(JSON.stringify(response.output.value, null, 2));
